ros/lib/utils.py

```diff
@@ -78,14 +79,43 @@ def user_data_from_identity(identity):
     return identity['user']
 
 
-def validate_ros_payload(is_ros, cloud_provider):
-    """
-    Validate ros payload.
-    :param is_ros: is_ros boolean flag
-    :param cloud_provider: cloud provider value
-    :return: True if cloud_provider is not none and is_ros flag is set to true, False otherwise.
-    """
-    return True if is_ros and cloud_provider is not None else False
+def validate_ros_payload(func):
+    def payload_validator(*args, **kwargs):
+        try:
+            payload_data = args[1]
+            if 'input' in payload_data.keys():  # input key is found only in engine msgs
+                caller_prefix = 'ENGINE CONSUMER'
+                account = payload_data['input']['host']['account']
+                org_id = payload_data['input']['platform_metadata']['org_id']
+                is_ros = payload_data["input"]["platform_metadata"].get("is_ros")
+                cloud_provider = payload_data["results"]["system"]["metadata"].get('cloud_provider')
+                system_id = payload_data["input"]["host"]["id"]
+            else:
+                caller_prefix = 'EVENTS CONSUMER'
+                account, org_id = payload_data['host']['account'], payload_data['host']['org_id']
+                system_id = payload_data['host']['id']
+                is_ros = payload_data["platform_metadata"].get("is_ros")
+                cloud_provider = payload_data['host']['system_profile'].get('cloud_provider')
+        except Exception as e:
+            LOG.error(e)
+            is_ros = False
+            cloud_provider, system_id, caller_prefix, account, org_id = None, None, None, None, None
+
+        is_ros_flag = True if is_ros in ['true', True, 'TRUE', '1'] else False
+        cloud_provider_flag = cloud_provider in [provider.value for provider in CloudProvider]
```

Comment:
and separate for this as well?

Reply:
I gave some thought about this one, post our discussion. IMO it would make sense for this native logic to remain directly inside the decorator, especially when it is not verbose.